odoo/addons/base/models/ir_http.py

```diff
@@ -150,9 +150,21 @@ def _slugify_one(cls, value: str, max_length: int = 0) -> str:
                 return slugify_lib.slugify(value, max_length=max_length)
             except TypeError:
                 pass
-        uni = unicodedata.normalize('NFKD', value).encode('ascii', 'ignore').decode('ascii')
-        slug_str = re.sub(r'[\W_]+', '-', uni).strip('-').lower()
-        return slug_str[:max_length] if max_length > 0 else slug_str
+        uni = unicodedata.normalize('NFKD', value)
+        slugified_segments = []
+        for slug in re.split('-|_| ', uni):
+            # Special symbols get replaced by -: ^h☺e$#!l(%l}o你好& becomes -h☺e---l--l-o你好-
+            slug = re.sub(r"[$&+,:;=?@#|'<>.^*()%!\-{}/\\\~]", '-', slug)
+            # Consecutive dashes, like ----, get replaced by a single dash: -h☺e---l--l-o你好- becomes -h☺e-l-l-o你好-
+            slug = re.sub(r'--+', '-', slug)
+            # Any character that is not a word or a dash gets removed: -h☺e-l-l-o你好- becomes -he-l-l-o你好-
+            slug = re.sub(r'([^\w-])+', '', slug)
+            # Leading and trailing dashes get removed: -he-l-l-o你好- becomes he-l-l-o你好
+            slug = slug.strip('-')
+            if (len(slug) > 0):
+                slugified_segments.append(slug.lower())
```

Comment:
Now that we support i18n, shall we keep calling lower? Genuine question, I don't have the answer.

Reply:
Several tests failed when I tried to remove .lower(), for safety I prefer keeping it